metrics/stackdriver_exporter.go

```diff
@@ -63,10 +91,18 @@ func newOpencensusSDExporter(o stackdriver.Options) (view.Exporter, error) {
 // TODO should be properly refactored to be able to inject the getMonitoredResourceFunc function.
 // 	See https://github.com/knative/pkg/issues/608
 func newStackdriverExporter(config *metricsConfig, logger *zap.SugaredLogger) (view.Exporter, error) {
-	gm := gcpMetadataFunc()
+	gm := getGCPMetadata(config)
 	mtf := getMetricTypeFunc(config.stackdriverMetricTypePrefix, config.stackdriverCustomMetricTypePrefix)
+	co, err := getStackdriverExporterOptions(&config.stackdriverClientConfig)
+	if err != nil {
+		logger.Warnw("Issue configuring Stackdriver exporter client options, no additional client options will be used: ", zap.Error(err))
+	}
+	// Automatically fall back on Google application default credentials
 	e, err := newStackdriverExporterFunc(stackdriver.Options{
-		ProjectID:               config.stackdriverProjectID,
+		ProjectID:               config.stackdriverClientConfig.ProjectID,
```

Comment:
I think it's better to use `gm.projectId` and `gm.location` to keep consistent in case `gm` has values fetched from GCE metadata service but config.stackdriverClientConfig.ProjectID is empty.

Reply:
Oh yeah, it is actually a bug not to return gm.projectID & gm.location. Good catch!